app/src/main/java/org/ea/sqrl/activites/create/SaveIdentityActivity.java

```diff
@@ -78,19 +80,29 @@ protected void onCreate(Bundle savedInstanceState) {
                     handler.post(() -> hideProgressPopup());
                 }
 
-                long newIdentityId = mDbHelper.newIdentity(SaveIdentityActivity.this, storage.createSaveData());
-                mDbHelper.updateIdentityName(SaveIdentityActivity.this, newIdentityId,
-                        txtIdentityName.getText().toString());
+                if(!mDbHelper.hasIdentities()) {
+                    firstIdentity = true;
+                }
 
+                long newIdentityId = mDbHelper.newIdentity(SaveIdentityActivity.this, storage.createSaveData());
                 SqrlApplication.saveCurrentId(this.getApplication(), newIdentityId);
 
                 handler.post(() -> {
-                    txtIdentityName.setText("");
                     txtNewPassword.setText("");
                     txtRetypePassword.setText("");
 
                     SaveIdentityActivity.this.finish();
-                    startActivity(new Intent(this, NewIdentityDoneActivity.class));
+
+                    Intent nextActivity = null;
+
+                    if(firstIdentity) {
+                        nextActivity = new Intent(this, NewIdentityDoneActivity.class);
```

Comment:
Should we add code to set the standard name here?

Reply:
Same as above:

> Since the previous PR, this happens already in `newIdentity()`, so it is already taken care of at that point.